src/internal/components/option/index.tsx

```diff
@@ -59,15 +60,17 @@ const Option = ({
     />
   );
 
+  // When generic props are assigned assistive technologies treat option nodes as semantic (generic) groups.
+  // See https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/generic_role
+  const genericGroupProps = isGenericGroup
```

Comment:
I've read the comment, the linked page, and the code a few times now, and still don't understand what this is doing. Perhaps it's too early on a Monday, but perhaps this could be expanded?

Reply:
maybe also expanding on _why_ (I think it's because sometimes these attributes are set on the parent instead?)